<?xml version="1.0"?>

<launch>


  <!-- FCU communication -->
  <node name="rosflight_io" pkg="rosflight" type="rosflight_io" output="screen">
    <param name="udp" value="false"/>
  </node>

  <!-- Load common parameters -->
  <rosparam command="load" file="$(find roscopter)/params/F450.yaml"/>

  <!-- PID Position Controller -->
  <node name="controller" pkg="roscopter" type="controller">
    <remap from="estimate" to="odom"/>
  </node>

  <!-- Estimator -->
  <rosparam command="load" file="$(find roscopter)/params/ekf_gains.yaml" ns="estimator"/>
  <rosparam command="load" file="$(find roscopter)/params/ekf_drone.yaml" ns="estimator"/>
  <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
    <remap from="imu" to="/imu/data"/>
    <remap from="truth/pose" to="/vrpn_client_node/planck/pose"/>
    <remap from="truth/transform" to="/multirotor/truth/NED"/>
  </node>

  <!-- Waypoint Manager -->
  <node name="waypoint_manager" pkg="roscopter" type="waypoint_manager.py" output="screen">
    <remap from="waypoint" to="high_level_command"/>
    <remap from="state" to="odom"/>
  </node>

</launch>
